components/TxForm.tsx

```diff
@@ -38,95 +57,138 @@ export const TxForm = () => {
-  const [recipient, setRecipient] = useState<string>(
-    '0zk1qyn0qa5rgk7z2l8wyncpynmydgj7ucrrcczhl8k27q2rw5ldvv2qrrv7j6fe3z53ll5j4fjs9j5cmq7mxsaulah7ykk6jwqna3nwvxudp5w6fwyg8cgwkwwv3g4'
-  );
+  const { data } = useTokenAllowance({ address: selectedToken?.address || '' });
+  const tokenAllowance =
+    tokenAllowances.get(selectedToken?.address || '') || data || BigNumber.from(0);
+  const [recipient, setRecipient] = useState<string>(recipientAddress || '');
   const needsApproval =
-    tokenAddress !== ethAddress &&
-    ethers.utils.parseUnits(tokenAmount || '0', tokenDecimals).gt(tokenAllowance);
+    selectedToken?.address !== ethAddress &&
+    ethers.utils.parseUnits(tokenAmount || '0', selectedToken?.decimals).gt(tokenAllowance);
+  const onSubmit = handleSubmit(async (values) => {
+    setRecipient(values.recipient);
+    setTokenAmount(values.amount);
+    onReviewOpen();
+  });
+
+  const updateOnNetworkChange = useCallback(
+    (net: GetNetworkResult) => {
+      if (net && net?.chain) {
+        const chain = networks[net.chain.id];
+        const token = buildBaseToken(chain.baseToken, net.chain.id);
+        setSelectedToken(token);
+        setValue('token', chain.baseToken.name);
+      }
+    },
+    [setValue]
+  );
+
+  useEffect(() => {
+    const unwatch = watchNetwork(updateOnNetworkChange);
+    return unwatch;
+  }, [updateOnNetworkChange]);
+
+  useEffect(() => {
+    if (!selectedToken) {
+      setSelectedToken(tokenList[0]);
+    }
+  }, [selectedToken, tokenList]);
 
   return (
     <Box width="24rem">
-      <FormControl>
-        <FormLabel>Recipient address</FormLabel>
-        <Input
-          type="string"
-          variant="outline"
-          size="lg"
-          pr="4.5rem"
-          height="4rem"
-          mb=".75rem"
-          value={recipient}
-          onChange={(e) => setRecipient(e.target.value)}
-        />
-      </FormControl>
-      <FormControl>
-        <FormLabel>Token</FormLabel>
-        <TokenInput
-          onSelect={(token) => {
-            const { address, decimals, symbol, name } = token;
-            setTokenAddress(address);
-            setTokenDecimals(decimals);
-            setTokenSymbol(symbol);
-            setTokenName(name);
-          }}
-        />
-      </FormControl>
-      <FormControl>
-        <FormLabel>Amount</FormLabel>
-        <InputGroup size="lg" width="auto" height="4rem">
+      <form onSubmit={onSubmit}>
+        <FormControl isInvalid={Boolean(errors.recipient?.message)}>
+          <FormLabel>Recipient address</FormLabel>
           <Input
-            type="number"
             variant="outline"
             size="lg"
             pr="4.5rem"
-            height="100%"
-            value={tokenAmount}
-            onChange={(e) => setTokenAmount(e.target.value)}
+            height="4rem"
+            mb=".25rem"
+            {...register('recipient', {
```

Comment:
looks great! nit: In this PR we may have hit the threshold for component logic in the jsx markup. let's call a refactor here "nice to have" if needed at all. not merge-blocking!